.github/workflows/check_code_style_pull_request.yml

```diff
@@ -0,0 +1,20 @@
+name: Code Style Pull Request
+
+on: [pull_request]
+
+
+jobs:
+  linter:
+    runs-on: ubuntu-latest
+    name: Lint flutter code
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Set up Flutter
+      uses: subosito/flutter-action@v1
+    - run: flutter pub get
+    - name: Analyze Flutter
+      uses: ValentinVignal/action-dart-analyze@v0.11
```

Comment:
Can you explain this file? more specifically this line?

Reply:
It is using this Github action to analyse dart cod:  https://github.com/marketplace/actions/dart-analyze.

When a pull request will be created, it will automatically post a comment if the code doesn't follow the rules you set in your `analysis_option.yml`. It will also highlight the issues in the `File changed` section of the PR.

Reply:
Sounds great!
Thanks for explaining